devito/symbolics/printer.py

```diff
@@ -159,8 +168,17 @@ def _print_Abs(self, expr):
         if isinstance(self.compiler, AOMPCompiler):
             return "fabs(%s)" % self._print(expr.args[0])
         # Check if argument is an integer
-        func = "abs" if has_integer_args(*expr.args[0].args) else "fabs"
-        return "%s(%s)" % (func, self._print(expr.args[0]))
+        if has_integer_args(*expr.args[0].args):
+            func = "abs"
+        elif self.single_prec(expr):
+            func = "fabsf"
+        elif any([isinstance(a, Real) for a in expr.args[0].args]):
+            # The previous condition isn't sufficient to detect case with
+            # Python `float`s in that case, fall back to the "default"
+            func = "fabsf" if self.single_prec() else "fabs"
```

Comment:
I can see two self.single_prec tests. Isn;t this redundant here?
Can it be one check?